src/benchmarkstt/output/core.py

```diff
@@ -0,0 +1,70 @@
+from benchmarkstt import output
+from benchmarkstt.schema import Schema
+from csv import writer
+import sys
+
+
+class ReStructuredText(output.Base):
+    def result(self, title, result):
+        print(title)
+        print('=' * len(title))
+        print()
+
+        if type(result) is float:
+            print("%.6f" % (result,))
+        else:
+            print(result)
+        print()
+
+
+class MarkDown(output.Base):
+    def result(self, title, result):
+        print('# %s' % (title,))
+        print()
+
+        if type(result) is float:
+            print("%.6f" % (result,))
+        else:
+            print(result)
+        print()
+
+
+class Json(output.Base):
+    def __init__(self):
+        self._line = None
+
+    def __enter__(self):
+        if self._line is not None:
+            raise ValueError("Already open")
+        print('{')
+        self._line = 0
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self._line = None
+        print('\n}')
+
+    def result(self, title, result):
+        if self._line != 0:
+            print(',')
+        self._line += 1
+        en = Schema.dumps
+        print('\t%s: %s' % (en(title), en(result)), end='')
+
+
+class Csv(output.Base):
```

Comment:
I think having a CSV output will produce more problems than it's worth. The nature of CSV is a flat table, but we're trying to add data that is more structured. The current output has the headers on the left column, when they should be on the top row, with the values underneath, but I don't know how you'd do a word-by-word comparison with that.

It might be worth removing support for this output format.

Reply:
The structured output format would be part of a field, e.g.

```
$> benchmarkstt bbc.ref.txt bbc.hyp.txt --config config.ini --wer --diffcounts --worddiffs html --output-format csv
```
gives
```
wer,0.16957622168862352
diffcounts,"OpcodeCounts(equal=13268, replace=1647, insert=472, delete=494)"
worddiffs," tonight the prime minister theresa may the leader of the conservative party and the leader of<span class=""insert""> the</span> labour party jeremy corbyn face the voters welcome<span class=""delete""> to question time</span> so over the next<span class=""insert""> ninety</span> minutes the leaders of the two larger parties are going to be quizzed by our audience here in york now this audience is made up like this<span class=""delete""> -</span> just<span class=""delete""> a third</span> say they intend to vote<span class=""delete""> conservative next week</span><span class=""insert""> conserve it</span> the same<span class=""delete""> number</span>[...TRUNCATED...]
```

Column 1 is always the metrics title, Column 2 the result of the metric, which can be whatever. It is mostly handy for example for inserting results in databases through command line...